vavr/pom.xml

```diff
@@ -20,6 +20,12 @@
     </licenses>
     <dependencies>
         <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
```

Comment:
What is lombok used for?

Reply:
Cuts out all the boilerplay code for `get`, `set` `hash` `equals` etc.

If I have a class like this:

```java
@EqualsAndHash
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Foo {

    private String name;
    private Integer age;

}
```

Then I get a class generated that has this:

```
Foo f = new Foo("Christian",1234);
String name = f.getName();
Integer age = f.getAge();
Set<Foo> foos = HashSet.of(f);
Foo baz = new Foo("Christian",1234);
System.out.println(foos.contains(baz)); // prints true because hash and equals are created
```

There are a lot of annotations. It really reduces code

Reply:
@chb0github we keep Vavr without dependencies on the third-part libraries

Reply:
Lombok would be especially problematic, since it's a _manipulating bytecode_, it's doing totally non-standard things, not just a regular annotation processor, hence it's incompatible with GWT, which Vavr supports.

Reply:
@ruslansennov - it's a `provided` dependency. Which means, it doesn't go with the deployment artifact. Meaning it is functionally no dependency at all.

Reply:
Lombok will hook into the internal compiler code of javac and ECJ and modify their internal in-memory AST representation to convince the compiler to generate bytecode according to the annotations you use. With the way it works, you will get into trouble as soon as you'll try to use the same code in another compilation environment, which requires consistency at the source level, since Lombok generated (byte-)code is invisible to those tools. While that may or may not work with GWT, since GWT also uses ECJ, it will definitely break with any other transpiler or tool that parses java source code. It's a total hack that can break any time also if the javac or ECJ team decides to change some internal implementation details of their compilers. Because of these, for me it's totally out of question to use it in any serious project.